integration-tests/src/test/java/org/openmrs/module/fhir2/providers/r3/AllergyIntoleranceFhirResourceProviderIntegrationTest.java

```diff
@@ -514,4 +514,39 @@ public void shouldSearchForExistingAllergyAsXML() throws Exception {
 		
 	}
 	
+	@Test
+	public void shouldReturnCountForAllergyIntoleranceAsJson() throws Exception {
+		MockHttpServletResponse response = get("/AllergyIntolerance?_summary=count").accept(FhirMediaTypes.JSON).go();
+		
+		assertThat(response, isOk());
+		assertThat(response.getContentType(), is(FhirMediaTypes.JSON.toString()));
+		assertThat(response.getContentAsString(), notNullValue());
+		
+		System.out.println(response.getContentAsString());
+		
+		Bundle result = readBundleResponse(response);
+		
+		assertThat(result, notNullValue());
+		assertThat(result.getType(), equalTo(Bundle.BundleType.SEARCHSET));
+		
+		assertThat(result, hasProperty("total", equalTo(6)));
```

Comment:
What we should also assert is the number of resources returned in the bundle. It should ideally be 1, as per our implementation. Thoughts?

Reply:
@varung-31 I believe _summary=count encodes the response in summary mode. So no resource is returned.
 So the size of the entry components would be 0.
The response as a string for the URL http://localhost:8080/openmrs/ws/fhir2/R4//Patient?_summary=count : 

```
{
    "resourceType": "Bundle",
    "id": "fb89f580-c729-4472-a2b2-06a3b22c5622",
    "meta": {
        "lastUpdated": "2021-03-22T13:49:48.385+05:30",
        "tag": [
            {
                "system": "http://terminology.hl7.org/CodeSystem/v3-ObservationValue",
                "code": "SUBSETTED",
                "display": "Resource encoded in summary mode"
            }
        ]
    },
    "type": "searchset",
    "total": 2
}
```
This is also the response to the URL : http://localhost:8080/openmrs/ws/fhir2/R4//Patient?_summary=count&_count=1